### PR TITLE
Detect duplicates between bioconda and other channels.

### PR DESCRIPTION
**Pull request by johanneskoester**
@bgruening @daler, this new subcommand allows us to detect duplicates between our channel and others.

Look at this list:
```
name	version	build	channels
aniso8601	1.1.0	py27_0	conda-forge
aniso8601	1.1.0	py34_0	conda-forge
aniso8601	1.1.0	py35_0	conda-forge
appdirs	1.4.0	py27_0	conda-forge
appdirs	1.4.0	py34_0	conda-forge
appdirs	1.4.0	py35_0	conda-forge
appdirs	1.4.0	py36_0	conda-forge
argh	0.26.2	py27_0	conda-forge
argh	0.26.2	py34_0	conda-forge
argh	0.26.2	py35_0	conda-forge
arrow	0.7.0	py27_0	conda-forge
arrow	0.7.0	py34_0	conda-forge
arrow	0.7.0	py35_0	conda-forge
bcdoc	0.16.0	py27_0	conda-forge
bcdoc	0.16.0	py34_0	conda-forge
bcdoc	0.16.0	py35_0	conda-forge
bcdoc	0.16.0	py36_0	conda-forge
blinker	1.4	py27_0	conda-forge
blinker	1.4	py34_0	conda-forge
blinker	1.4	py35_0	conda-forge
blinker	1.4	py36_0	conda-forge
bottle	0.12.9	py27_0	conda-forge
bottle	0.12.9	py34_0	conda-forge
bottle	0.12.9	py35_0	conda-forge
bottle	0.12.9	py36_0	conda-forge
brewer2mpl	1.4.1	py27_0	conda-forge
brewer2mpl	1.4.1	py34_0	conda-forge
brewer2mpl	1.4.1	py35_0	conda-forge
expat	2.1.0	0	conda-forge
fftw	3.3.4	0	conda-forge
flask-basicauth	0.2.0	py27_0	conda-forge
flask-basicauth	0.2.0	py34_0	conda-forge
flask-basicauth	0.2.0	py35_0	conda-forge
flask-script	2.0.5	py27_0	conda-forge
flask-script	2.0.5	py34_0	conda-forge
flask-script	2.0.5	py35_0	conda-forge
flask-sqlalchemy	2.1	py27_0	conda-forge
flask-sqlalchemy	2.1	py34_0	conda-forge
flask-sqlalchemy	2.1	py35_0	conda-forge
functools32	3.2.3.2	py27_0	conda-forge
graphviz	2.38.0	3	conda-forge
graphviz	2.38.0	4	conda-forge
grequests	0.3.0	py27_0	conda-forge
httplib2	0.9.2	py27_0	conda-forge
httplib2	0.9.2	py34_0	conda-forge
httplib2	0.9.2	py35_0	conda-forge
isodate	0.5.4	py27_0	conda-forge
isodate	0.5.4	py34_0	conda-forge
isodate	0.5.4	py35_0	conda-forge
isodate	0.5.4	py36_0	conda-forge
jmespath	0.9.0	py27_0	conda-forge
jmespath	0.9.0	py34_0	conda-forge
jmespath	0.9.0	py35_0	conda-forge
keepalive	0.5	py27_0	conda-forge
keepalive	0.5	py34_0	conda-forge
keepalive	0.5	py35_0	conda-forge
keepalive	0.5	py36_0	conda-forge
lemon	1.3.1	0	conda-forge
libssh2	1.7.0	0	conda-forge
m4	1.4.17	0	conda-forge
mmtf-python	1.0.5	py27_0	conda-forge
mmtf-python	1.0.5	py34_0	conda-forge
mmtf-python	1.0.5	py35_0	conda-forge
mmtf-python	1.0.5	py36_0	conda-forge
multiprocess	0.70.4	py27_0	conda-forge
multiprocess	0.70.4	py34_0	conda-forge
multiprocess	0.70.4	py35_0	conda-forge
ndg-httpsclient	0.4.2	py27_0	conda-forge
ndg-httpsclient	0.4.2	py34_0	conda-forge
ndg-httpsclient	0.4.2	py35_0	conda-forge
ndg-httpsclient	0.4.2	py36_0	conda-forge
netcdf-fortran	4.4.4	0	conda-forge
nglview	0.6.2.2	py27_0	conda-forge
nglview	0.6.2.2	py35_0	conda-forge
nglview	0.6.2.3	py27_0	conda-forge
nglview	0.6.2.3	py35_0	conda-forge
pathtools	0.1.2	py27_0	conda-forge
pathtools	0.1.2	py34_0	conda-forge
pathtools	0.1.2	py35_0	conda-forge
pathtools	0.1.2	py36_0	conda-forge
pyaml	15.8.2	py27_0	conda-forge
pyaml	15.8.2	py34_0	conda-forge
pyaml	15.8.2	py35_0	conda-forge
pyaml	15.8.2	py36_0	conda-forge
pyasn1-modules	0.0.8	py27_0	conda-forge
pyasn1-modules	0.0.8	py34_0	conda-forge
pyasn1-modules	0.0.8	py35_0	conda-forge
pycodestyle	2.0.0	py27_0	conda-forge
pycodestyle	2.0.0	py34_0	conda-forge
pycodestyle	2.0.0	py35_0	conda-forge
pydoe	0.3.8	py27_0	conda-forge
pydoe	0.3.8	py34_0	conda-forge
pydoe	0.3.8	py35_0	conda-forge
pydoe	0.3.8	py36_0	conda-forge
pydotplus	2.0.2	py27_0	conda-forge
pydotplus	2.0.2	py34_0	conda-forge
pydotplus	2.0.2	py35_0	conda-forge
pydotplus	2.0.2	py36_0	conda-forge
pyfiglet	0.7.5	py27_0	conda-forge
pyfiglet	0.7.5	py35_0	conda-forge
pyfiglet	0.7.5	py36_0	conda-forge
pyinotify	0.9.6	py27_0	conda-forge
pyinotify	0.9.6	py34_0	conda-forge
pyinotify	0.9.6	py35_0	conda-forge
pyinotify	0.9.6	py36_0	conda-forge
pytest-cov	2.4.0	py27_0	conda-forge
pytest-cov	2.4.0	py34_0	conda-forge
pytest-cov	2.4.0	py35_0	conda-forge
pytest-cov	2.4.0	py36_0	conda-forge
pytest-mock	1.1	py27_0	conda-forge
pytest-mock	1.1	py34_0	conda-forge
pytest-mock	1.1	py35_0	conda-forge
pytest-mpl	0.5	py27_0	conda-forge
pytest-mpl	0.5	py34_0	conda-forge
pytest-mpl	0.5	py35_0	conda-forge
pytest-xdist	1.14	py27_0	conda-forge
pytest-xdist	1.14	py34_0	conda-forge
pytest-xdist	1.14	py35_0	conda-forge
python-dateutil	2.6.0	py27_0	conda-forge
python-dateutil	2.6.0	py34_0	conda-forge
python-dateutil	2.6.0	py35_0	conda-forge
python-dateutil	2.6.0	py36_0	conda-forge
python-igraph	0.7.1.post6	py27_1	conda-forge
python-igraph	0.7.1.post6	py27_2	conda-forge
python-igraph	0.7.1.post6	py34_1	conda-forge
python-igraph	0.7.1.post6	py35_1	conda-forge
python-igraph	0.7.1.post6	py35_2	conda-forge
python-igraph	0.7.1.post6	py36_2	conda-forge
r-bh	1.60.0_2	r3.3.1_0	r
r-bit	1.1_12	r3.3.1_0	r
r-bit	1.1_12	r3.3.2_0	r	conda-forge
r-bit64	0.9_5	r3.3.1_0	r
r-bit64	0.9_5	r3.3.2_0	r	conda-forge
r-car	2.1_4	r3.3.2_0	r	conda-forge
r-checkmate	1.8.2	r3.3.2_0	conda-forge
r-data.table	1.10.0	r3.3.1_0	r
r-ddrtree	0.1.5	r3.3.2_0	conda-forge
r-densityclust	0.2.1	r3.3.2_0	conda-forge
r-digest	0.6.9	r3.3.1_0	r
r-dplyr	0.5.0	r3.3.1_0	r
r-dt	0.1	r3.3.1_0	r
r-fastica	1.2_0	r3.3.2_0	conda-forge
r-findpython	1.0.1	r3.3.2_0	r
r-getopt	1.20.0	r3.3.2_0	r	conda-forge
r-ggplot2	2.1.0	r3.3.1_0	r
r-ggplot2	2.2.0	r3.3.1_0	r
r-ggplot2	2.2.1	r3.3.2_0	conda-forge
r-gsubfn	0.6_6	r3.3.2_0	conda-forge
r-inline	0.3.14	r3.3.1_0	r
r-inline	0.3.14	r3.3.2_0	r	conda-forge
r-logging	0.7_103	r3.3.1_0	r
r-logging	0.7_103	r3.3.2_0	r	conda-forge
r-qlcmatrix	0.9.5	r3.3.2_0	conda-forge
r-r.methodss3	1.7.1	r3.3.1_0	r
r-r.methodss3	1.7.1	r3.3.2_0	r	conda-forge
r-r.oo	1.21.0	r3.3.1_0	r
r-r.oo	1.21.0	r3.3.2_0	r	conda-forge
r-r.utils	2.5.0	r3.3.1_0	r
r-r.utils	2.5.0	r3.3.2_0	r	conda-forge
r-reshape2	1.4.2	r3.3.1_0	r
r-reshape2	1.4.2	r3.3.2_0	r	conda-forge
r-rjson	0.2.15	r3.3.1_0	r
r-rjson	0.2.15	r3.3.2_0	r	conda-forge
r-rlist	0.4.6.1	r3.3.1_0	r
r-rlist	0.4.6.1	r3.3.2_0	r	conda-forge
r-rmysql	0.10.11	r3.3.2_0	conda-forge
r-rocr	1.0_7	r3.3.1_0	r
r-rocr	1.0_7	r3.3.2_0	r	conda-forge
r-scales	0.4.1	r3.3.1_0	r
r-sqldf	0.4_10	r3.3.2_0	conda-forge
r-stringr	1.1.0	r3.3.1_0	r
r-testthat	1.0.2	r3.3.1_0	r
r-tibble	1.2	r3.3.1_0	r
r-vgam	1.0_2	r3.3.1_0	r
r-vgam	1.0_2	r3.3.2_0	r
r-viridislite	0.1.3	r3.3.1_0	r
r-viridislite	0.1.3	r3.3.2_0	r
rdflib	4.2.1	py27_0	conda-forge
rdflib	4.2.1	py35_0	conda-forge
rdflib	4.2.2	py27_0	conda-forge
rdflib	4.2.2	py35_0	conda-forge
rdflib	4.2.2	py36_0	conda-forge
ruamel.ordereddict	0.4.6	py27_0	conda-forge
ruamel.yaml	0.11.11	py27_1	conda-forge
ruamel.yaml	0.11.11	py34_1	conda-forge
ruamel.yaml	0.11.11	py35_1	conda-forge
sed	4.2.2	0	conda-forge
setuptools_scm	1.11.1	py27_0	conda-forge
setuptools_scm	1.11.1	py34_0	conda-forge
setuptools_scm	1.11.1	py35_0	conda-forge
sortedcontainers	1.5.3	py27_0	conda-forge
sortedcontainers	1.5.3	py34_0	conda-forge
sortedcontainers	1.5.3	py35_0	conda-forge
sortedcontainers	1.5.3	py36_0	conda-forge
sparqlwrapper	1.7.6	py27_0	conda-forge
sparqlwrapper	1.7.6	py34_0	conda-forge
sparqlwrapper	1.7.6	py35_0	conda-forge
sparqlwrapper	1.7.6	py36_0	conda-forge
sparsehash	2.0.3	0	conda-forge
sphinx-argparse	0.1.15	py27_0	conda-forge
sphinx-argparse	0.1.15	py34_0	conda-forge
sphinx-argparse	0.1.15	py35_0	conda-forge
suds-jurko	0.6	py27_0	conda-forge
suds-jurko	0.6	py27_1	conda-forge
suds-jurko	0.6	py34_1	conda-forge
suds-jurko	0.6	py35_0	conda-forge
suds-jurko	0.6	py35_1	conda-forge
suds-jurko	0.6	py36_1	conda-forge
tabulate	0.7.5	py27_0	conda-forge
tabulate	0.7.5	py34_0	conda-forge
tabulate	0.7.5	py35_0	conda-forge
tqdm	4.7.2	py27_0	conda-forge
tqdm	4.7.2	py34_0	conda-forge
tqdm	4.7.2	py35_0	conda-forge
typing	3.5.0.1	py27_0	conda-forge
typing	3.5.0.1	py34_0	conda-forge
typing	3.5.0.1	py35_0	conda-forge
uritemplate	0.6	py27_0	conda-forge
uritemplate	0.6	py34_0	conda-forge
uritemplate	0.6	py35_0	conda-forge
watchdog	0.8.3	py27_0	conda-forge
watchdog	0.8.3	py34_0	conda-forge
watchdog	0.8.3	py35_0	conda-forge
ws4py	0.3.2	py27_0	conda-forge
ws4py	0.3.2	py34_0	conda-forge
ws4py	0.3.2	py35_0	conda-forge
xgboost	0.6a2	py27_0	conda-forge
xgboost	0.6a2	py35_0	conda-forge
xgboost	0.6a2	py36_0	conda-forge
yasm	1.3.0	0	conda-forge
```

I will create a PR to bioconda-recipes that removes these recipes and also delete them from anaconda. We can safely do that because we have exactly the same version.

### CODE REVIEW

**Comment by bgruening** [comment]
@johanneskoester nice! So this list is the strict version?

**Comment by johanneskoester** [comment]
Yes, down to the build number. See the corresponding PR in bioconda-recipes.

File: bioconda_utils/cli.py [hunk 1]
```diff
@@ -96,6 +97,46 @@ def select_recipes(packages, git_range, recipe_folder, config_filename, config,
...
+        print('name', 'channels', sep='\t')
+
+    def get_packages(channel):
+        return {get_spec(pkg)
+                for repodata in utils.get_channel_repodata(channel)
+                for pkg in repodata['packages'].values()}
+
+    channels = config['channels']
+
+    packages = get_packages(channels[0])
```
Comment (bgruening) [inline bioconda_utils/cli.py]: This line can be removed.

Comment (johanneskoester) [inline bioconda_utils/cli.py]: Thx!

Comment (bgruening) [inline bioconda_utils/cli.py]: My fault.

**Pull request merged**